src/main/java/com/datalogics/pdf/samples/manipulation/RemoveInteractivity.java

```diff
@@ -127,6 +128,10 @@ public static void removeInteractivity(final URL inputUrl, final URL outputUrl)
         try {
             pdfDoc = DocumentUtils.openPdfDocument(inputUrl);
 
+            // Before removing any interactive elements, generate their appearances using the AppearanceService
+            // to ensure that the document has all required appearances so the document looks correct at the end.
+            AppearanceService.generateAppearances(pdfDoc, null, null);
```

Comment:
@datalogics-pgallot, the code in the sample would then have to construct and add to the _APContext_ all the pieces for getting barcodes to generate. By passing **null** here instead, we let PDF Java Toolkit Core do that work.